object-manipulation-2/src/defaults.js

```diff
@@ -0,0 +1,19 @@
+/* exported defaults */
+// create storage for a copy of source and assign
+// look at each keys in source
+//  look at each key in target
+//    if keys matches key
+//      delete source at keys
+//  assign all properties and values from source to target.
+
+function defaults(target, source) {
+  var sourceCopy = Object.assign({}, source);
+  for (var keys in sourceCopy) {
+    for (var key in target) {
+      if (keys === key) {
+        delete source[keys];
```

Comment:
This is still deleting from `source`, which is a no-no. You should be doing your deleting on `courceCopy`, that is the safe object to delete from.